src/toxcore/dht/server.rs

```diff
@@ -414,21 +483,23 @@ impl Server {
     */
     fn handle_nat_ping_req(&self, packet: DhtRequest, payload: NatPingRequest, addr: SocketAddr) -> IoFuture<()> {
         let state = self.state.read();
-        let client = self.create_client(&addr, packet.spk, &state.peers_cache);
 
         if packet.rpk == self.pk { // the target peer is me
-            let resp_payload = NatPingResponse {
+            let resp_payload = DhtRequestPayload::NatPingResponse(NatPingResponse {
```

Comment:
Do we have to send nat ping response to ourselves? May be we should not even send nat ping requests to us?

Reply:
Here we check if someone else sent request to us. If so we handle this request, otherwise - resend to target peer.
It's not correct to check it here though, I've fixed it in other branch.